cookbook/upgrade/major_version.rst

```diff
@@ -0,0 +1,114 @@
+.. index::
+    single: Upgrading; Major Version
+
+Upgrading a Major Version (e.g. 2.7.0 to 3.0.0)
+===============================================
+
+Every few years, Symfony releases a new major version release (the first number
+changes). These releases are the trickiest to upgrade, as they are allowed to
+contain BC breaks. However, Symfony tries to make this upgrade process as
+smooth as possible.
+
+This means that you can update most of your code before the major release is
+actually released. This is called making your code *future compatible*.
+
+There are a couple of steps to upgrading a major version:
+
+#. :ref:`Make your code deprecation free <upgrade-major-symfony-deprecations>`;
+#. :ref:`Update to the new major version via Composer <upgrade-major-symfony-composer>`.
+#. :ref:`Update your code to work with the new version <upgrade-major-symfony-after>`
+
+.. _upgrade-major-symfony-deprecations:
+
+1) Make your Code Deprecation Free
+----------------------------------
+
+During the lifecycle of a major release, new features are added and method
+signatures and public API usages are changed. However, minor versions should
+not contain any backwards compatibility changes. It is made sure that there is
+a so-called *backwards compatibility layer* (or BC layer). This means that the
+old API will still work, while the new feature is used internally. This BC
+layer is then marked as *deprecated*, indicating that it will be
+removed/changed in the future.
+
+The major version is the only time all existing BC layers are removed. However,
+if you make sure you've fixed all deprecated usages in the last version of the
+previous major version, you should be able to upgrade to the new major version
+without problems. To help you with this, the last minor releases will trigger
+deprecated notices. For example, 2.7 and 2.8 trigger deprecated notices and if
+you do not have any notice while using 2.8, you can savely upgrade to 3.0.
```

Comment:
A slightly different suggestion, please mix and match - this stuff is just very important (which is why I'm so glad you wrote all this):

```rst
When the major version is released (e.g. 3.0.0), all deprecated features and functionality
are removed. So, as long as you've updated your code to stop using these deprecated
features in the last version before the major (e.g. 2.8.*), you should be able to upgrade
without a problem.

To help with this, the last minor releases will trigger deprecation notices. For example, Symfony
2.7 and 2.8 trigger deprecated notices that you can see in the web debug toolbar. When visiting
your application...
```